nucleus/payara-modules/hazelcast-bootstrap/src/main/java/fish/payara/nucleus/hazelcast/HazelcastCore.java

```diff
@@ -555,6 +561,26 @@ public int getPort() {
 
     @Override
     public UnprocessedChangeEvents changed(PropertyChangeEvent[] pces) {
-        return null;
+        List<UnprocessedChangeEvent> unprocessedChanges = new ArrayList<>();
+        for (PropertyChangeEvent pce : pces) {
+            if (pce.getPropertyName().equalsIgnoreCase("datagrid-encryption-enabled")) {
+                datagridEncryptionChanged = true;
+                unprocessedChanges.add(new UnprocessedChangeEvent(pce, "Hazelcast encryption settings changed"));
+            }
+        }
+
+        if (unprocessedChanges.isEmpty()) {
+            return null;
+        }
+        return new UnprocessedChangeEvents(unprocessedChanges);
+    }
+
+    public boolean isDatagridEncryptionEnabled() {
+        // If the datagrid encryption settings have changed, we want to return the opposite of the current setting
+        // to prevent the server changing encryption behaviour without a restart
+        if (datagridEncryptionChanged) {
+            return !configuration.getDatagridEncryptionEnabled().equalsIgnoreCase("true");
+        }
+        return  configuration.getDatagridEncryptionEnabled().equalsIgnoreCase("true");
```

Comment:
>  I don't get who a method called isDatagridEncryptionEnabled would return true when the configuration is false. That does not make sense for me even if the encryption has changed.

I tried to explain why in the comment but I'll expand.
If you've changed the settings (let's say you've enabled it), this will be replicated to both the domain.xml and the config bean - any call to `HazelcastRuntimeConfiguration.getDatagridEncryptionEnabled` will return true. However, we don't want to suddenly start demanding that everything be encrypted otherwise we'll end up with a clustered store where only _some_ things are encrypted. So until restart, we want to return the opposite.

Although now thinking about it again, the current impl is flawed in that if you enable it and then disable it again without restarting the server, it'll start saying it's enabled - I'll change it so it stores the old value as it's changed and returns that.

Reply:
That sounds more reasonable. As I understand it this should always return the initial value at start.